engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -272,12 +273,22 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
    */
   private def generateCollectorOutput(collector: CollectorKey, shards: Iterable[BackendCallKey]): Try[CallOutputs] = Try {
     val shardsOutputs = shards.toSeq sortBy { _.index.fromIndex } map { e =>
-      fetchCallOutputEntries(e) map { _.outputs } getOrElse(throw new RuntimeException(s"Could not retrieve output for shard ${e.scope} #${e.index}"))
+      fetchCallOutputs(e) getOrElse(throw new RuntimeException(s"Could not retrieve output for shard ${e.scope} #${e.index}"))
     }
     collector.scope.task.outputs map { taskOutput =>
-      val wdlValues = shardsOutputs.map(s => s.getOrElse(taskOutput.name, throw new RuntimeException(s"Could not retrieve output ${taskOutput.name}")))
-      val arrayOfValues = new WdlArray(WdlArrayType(taskOutput.wdlType), wdlValues)
-      taskOutput.name -> CallOutput(arrayOfValues, workflow.hash(arrayOfValues))
+      val callOutputs = shardsOutputs.map(s => s.getOrElse(taskOutput.name, throw new RuntimeException(s"Could not retrieve output ${taskOutput.name}")))
+      val arrayOfValues = new WdlArray(WdlArrayType(taskOutput.wdlType), callOutputs map { _.wdlValue })
+      val hashes = callOutputs map { output => {
+        output.hash orElse workflow.hash(output.wdlValue)
```

Comment:
Yeah I don't think that should ever happen, if there's no hash it should be because `writeToCache` is false, which means `workflow.hash(output.wdlValue)` will not compute the hash either this time.